File: Examples/test-suite/python/cpp11_variadic_function_templates_runme.py
from cpp11_variadic_function_templates import *

ec = EmplaceContainer()
ec.emplace(A())
ec.emplace(A(), B())
ec.emplace(A(), B(), C())
ec.emplace(A(), B(), C(), D())

def check(expected, got):
    if expected != got:
        raise RuntimeError("failed: {} != {}".format(expected, got))

a = A()
b = B()
c = C()
check(variadicmix1(), 20)
check(variadicmix1(a), 20)
check(variadicmix1(a, b), 10)
check(variadicmix1(a, b, c), 20)
check(variadicmix1(11, 22), 10)

File: Examples/test-suite/javascript/cpp11_variadic_function_templates_runme.js
var cpp11_variadic_function_templates = require("cpp11_variadic_function_templates");

const {A, B, C, D, variadicmix1} = cpp11_variadic_function_templates;

ec = new cpp11_variadic_function_templates.EmplaceContainer();
ec.emplace(new A());
ec.emplace(new A(), new B());
ec.emplace(new A(), new B(), new C());
ec.emplace(new A(), new B(), new C(), new D());

function check(expected, got) {
    if (expected != got) {
        throw new Error("failed: {} != {}".format(expected, got));
    }
}
a = new A();
b = new B();
c = new C();
check(variadicmix1(), 20);
check(variadicmix1(a), 20);
check(variadicmix1(a, b), 10);
check(variadicmix1(a, b, c), 20);
check(variadicmix1(11, 22), 10);
